[CUBRIDMAN-145] Restore shard-related content in manual

Kang-dot opened the pull request:
http://jira.cubrid.org/browse/CUBRIDMAN-145

Purpose

- Restore shard-related content in latest menual

Implementation

- N/A

Remarks

- Most of contents is taken from 9.3.0 version manual
- Add shard proxy log policy in shard log part

Kang-dot committed:
[CUBRIDMAN-145] restore and edit shard.rst file

- restore shard.rst from old commit
- add shard proxy log level policy content
- All changes apply to both en and ko

Kang-dot committed:
[CUBRIDMAN-145] Remove mysql content

- Remove mysql-relate content in shard.rst

Kang-dot committed:
[CUBRIDMAN-145] Add shard content in config.rst and control.rst

- add deleted shard content in config.rst and control.rst
- add proxy log level policy content in config.rst
- all changes apply to both en and ko

Kang-dot committed:
[CUBRIDMAN-145] Add shard.rst in toctree

- add shard.rst in toctree

kisoo-han reviewed (approved):
Thank you for your efforts!!

airnet73 commented on ko/admin/control.rst:
```diff
@@ -937,10 +942,15 @@ CCI 드라이버를 사용하여 C로 프로그램을 작성할 때는 에러 
     ++ cubrid broker is running.
 
+.. warning::
+    On Linux systems, the number of file descriptors (fd) required to run shards is moderately higher than SHARD_MAX_CLIENTS set in subrid_broker.conf, so when limiting the number of fds with "ulimit -n", it should be set appropriately larger than SHARD_MAX_CLIENTS.
```
It seems like an unnecessary sentence.

Kang-dot replied:
I made a mistake while adding content. I'll fix it right away.

airnet73 commented on ko/admin/config.rst:
```diff
@@ -2351,6 +2351,43 @@ cubrid_broker.conf 설정 파일과 기본 제공 파라미터
 |                                 |                         | SQL_LOG                         | string | ON                           | 가능      |
 |                                 |                         +---------------------------------+--------+------------------------------+-----------+
 |                                 |                         | SQL_LOG_MAX_SIZE                | KB     | 10000                        | 가능      |
+|                                 | 샤드                    | SHARD                           | string | OFF                          |           |
```
There is no dividing line between logs and shards.

Kang-dot replied:
I'll add it right away

airnet73 commented on ko/admin/control.rst:
```diff
@@ -1434,11 +1733,369 @@ HA 환경에서 브로커의 **ACCESS_MODE**\를 Read Only로 변경하고 해
 
 .. _broker-logs:
```
Looks like it needs to be deleted.  ".. _broker-logs:"

Kang-dot replied:
i will delete it

Kang-dot committed:
[CUBRIDMAN-145] modify ko/admin/control.rst

- rewrite english content to korean content
- remove misplaced elements

Kang-dot committed:
[CUBRIMAN0145] Add divide line in ko/admin/config.rst

- add divide line between log and shard

kwonhoil commented on ko/admin/control.rst:
```diff
@@ -93,6 +93,7 @@ CUBRID 브로커 프로세스를 제어하기 위한 **cubrid** 유틸리티 구
                |on <broker_name> |off <broker_name>
                |reset broker_name 
                |info
+               |getid -b <broker_name> [-f] shard_key
```
"브로커 제어" 설명 부분에 아래내용 참고하여 누락된 내용 추가. ( 9.3 메뉴얼 참고 )

CUBRID 브로커 프로세스를 제어하기 위한 cubrid 유틸리티 구문은 다음과 같다. <command>로 올 수 있는 명령어는 브로커 프로세스 구동을 위한 start , 종료를 위한 stop, 재시작을 위한 restart, 상태 확인을 위한 status, 브로커 접속 제한을 위한 acl, 명시한 브로커만 사용 가능하게 하거나 불가능하게 하는 on/off, 브로커 접속을 리셋하기 위한 reset, 설정 정보 출력을 위한 info, SHARD key를 가지고 SHARD ID(SHARD 데이터베이스 ID)를 얻는 getid가 있다.

Kang-dot replied:
I'll edit korean and english content in next commit

kwonhoil commented on ko/admin/control.rst:
```diff
@@ -994,12 +1004,32 @@ cubrid broker status에서 사용하는 [options]는 다음과 같다. 이들 
 
     작업 큐에 대기 중인 작업을 출력한다.
 
+.. option:: -c
```
cubrid broker status 설명 부분에 아래내용 참고하여 누락된 내용 추가. ( 9.3 메뉴얼 참고 )

cubrid broker status에서 사용하는 [options]는 다음과 같다. 이들 중 -b, -q, -c, -m, -S, -P, -f는 출력할 정보를 정의하는 모니터링 옵션이고, -s, -l, -t는 출력을 제어하는 옵션이다. 또한, -c, -m, -S, -P는 주로 SHARD 기능을 사용할 때 적용하는 옵션이다. 이 모든 옵션들은 서로 조합하여 사용하는 것이 가능하다.

Kang-dot replied:
I'll edit korean and english content in next commit

kwonhoil commented on en/admin/control.rst:
```diff
@@ -1438,13 +1741,370 @@ Broker configuration information
+     
+.. option:: -u DB_USER 
+
+    Specifies the DB account of the test target. When this option is omitted, the value of SHARD_DB_USER parameter is used if the value of a SHARD parameter in cubrid_broker.conf is ON. If the value of a SHARD parameter is OFF, "public" is input in CUBRID, or "root" is input in MySQL.
```
Please delete MySQL-related content.

Kang-dot replied:
I'll delete MySQL-relate content in next commit

kwonhoil commented on en/admin/control.rst:
```diff
@@ -1438,13 +1741,370 @@ Broker configuration information
+     
+.. option:: -p DB_PASSWORD 
+
+    Specifies the DB password of the test target. When this option is omitted, the value of SHARD_DB_PASSWORD parameter is used if the value of a SHARD parameter in cubrid_broker.conf is ON. If the value of a SHARD parameter is OFF, an empty string("") is input in CUBRID and MySQL.
```
Please delete MySQL-related content.

Kang-dot replied:
I'll delete MySQL-relate content in next commit

kwonhoil commented on ko/shard.rst:
```diff
@@ -0,0 +1,943 @@
+
+**auto increment는 각 shard DB 내에서만 유효**
+
+auto increment 속성 또는 SERIAL 등의 값이 각 shard DB 내에서만 유효하므로, 의도한 것과 다른 값을 반환할 수 있다.
```
* 아래 내용은 QA팀에서 발견하였고, 선별회의 결과 메뉴얼의 제약사항으로 수정하기로 결정 된 사항입니다.


SAHRD 힌트구문 작성된 DDL 구문은 지원 안함
shard_id(0)은 정상적으로 처리되며, shard_id(1)부터는 에러.
SHARD 구성 환경에서 스키마 생성 및 변경 등의 DDL구문은 shard 힌트가 지원 안 됨으로 각각의 shard DB에 접속하여 스키마 생성 및 변경을 처리하여야 한다.

오류 예시) 
   CREATE TABLE foo (col1 INT NOT NULL) /*+ SHARD_ID(1) */
   DROP TABLE IF EXISTS foo /*+ SHARD_ID(1) */

Kang-dot replied:
Add content as below

### korean version

**SHARD 힌트 구문이 작성된 DDL 구문은 지원 안함**

SHARD 구성 환경에서 스키마 생성 및 변경 등의 DDL구문은 SHARD 힌트가 지원 되지 않으므로 각각의 SHARD DB에 접속하여 스키마 생성 및 변경을 처리하여야 한다. shard_id(0)은 정상적으로 처리되며, shard_id(1)부터는 에러를 발생시킨다. 

오류 예시)

::

    CREATE TABLE foo (col1 INT NOT NULL) /*+ SHARD_ID(1) /

::

    DROP TABLE IF EXISTS foo /+ SHARD_ID(1) */

### english version

**DDL syntax with SHARD hint syntax not supported**

Since DDL phrases such as schema creation and change in the SHARD configuration environment do not support SHARD hints, each SHARD DB must be accessed to process schema creation and change. The shard_id (0) is normally processed, and an error is generated from the shard_id (1).

Error example)

::

    CREATE TABLE foo (col1 INT NOT NULL) /*+ SHARD_ID(1) /

::

    DROP TABLE IF EXISTS foo /+ SHARD_ID(1) */

Kang-dot committed:
[CUBRIDMAN-145] Remove MySQL-relate content and edit content

- remote MySQL-relate content
- edit controlling broker part
- edit Checking Broker Status

Kang-dot committed:
[CUBRIDMAN-145] Add Constraint

- add constraint about SHARD hint DDL
